public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/MetricNameField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+
+import { InlineField, Select } from '@grafana/ui';
+import { findOption, MetricsQueryEditorFieldProps, Options, toOption } from '../common';
+
+const MetricName: React.FC<MetricsQueryEditorFieldProps> = ({
+  query,
+  datasource,
+  subscriptionId,
+  replaceTemplateVariable,
+  onChange,
+}) => {
+  const [options, setOptions] = useState<Options>([]);
+  const azureMonitorIsConfigured = datasource.azureMonitorDatasource.isConfigured();
```

Comment:
I know this check wasn't your idea, but it seems to be repeated a lot and I wonder what the point is of checking if the subscription id is set. Isn't it better to just let the query to azure fail if the sub id is not provided?

Reply:
Yeah this is fair, ill modify this to do essentially the same thing, but with less duplication.